app/models/student.rb

```diff
@@ -250,6 +250,14 @@ def released_result_for?(assessment)
     end
   end
 
+  def grace_credits_used_for(assessment)
+    grouping = accepted_groupings.find_by(assessment_id: assessment.id)  # Find the grouping for this assessment
```

Comment:
The querying here is good, but you can make better use of existing methods, in particular `accepted_grouping_for`.

There may also be a `Grouping` instance method that you can use to get the number of grace credits used by the grouping. Note that in a grouping, all members have the same grace credit deductions.